wayland-client/examples/simple_layershell.rs

```diff
@@ -0,0 +1,289 @@
+    }
+
+    while state.running {
+        event_queue.blocking_dispatch(&mut state).unwrap();
+    }
+}
+
+struct State {
+    running: bool,
+    base_surface: Option<wl_surface::WlSurface>,
+    layer_shell: Option<zwlr_layer_shell_v1::ZwlrLayerShellV1>,
+    layer_surface: Option<zwlr_layer_surface_v1::ZwlrLayerSurfaceV1>,
+    buffer: Option<wl_buffer::WlBuffer>,
+    wm_base: Option<xdg_wm_base::XdgWmBase>,
+}
+
+impl Dispatch<wl_registry::WlRegistry, ()> for State {
+    fn event(
+        state: &mut Self,
+        registry: &wl_registry::WlRegistry,
+        event: wl_registry::Event,
+        _: &(),
+        _: &Connection,
+        qh: &QueueHandle<Self>,
+    ) {
+        if let wl_registry::Event::Global { name, interface, version } = event {
+            if interface == zwlr_layer_shell_v1::ZwlrLayerShellV1::interface().name {
+                let wl_layer = registry.bind::<zwlr_layer_shell_v1::ZwlrLayerShellV1, _, _>(
+                    name,
+                    version,
+                    qh,
+                    (),
+                );
+                state.layer_shell = Some(wl_layer);
+            } else if interface == wl_compositor::WlCompositor::interface().name {
+                let compositor =
+                    registry.bind::<wl_compositor::WlCompositor, _, _>(name, version, qh, ());
+                let surface = compositor.create_surface(qh, ());
+                state.base_surface = Some(surface);
+            } else if interface == wl_shm::WlShm::interface().name {
+                let shm = registry.bind::<wl_shm::WlShm, _, _>(name, version, qh, ());
+
+                let (init_w, init_h) = (3200, 240);
+
+                let mut file = tempfile::tempfile().unwrap();
+                draw(&mut file, (init_w, init_h));
+                let pool = shm.create_pool(file.as_raw_fd(), (init_w * init_h * 4) as i32, qh, ());
+                let buffer = pool.create_buffer(
+                    0,
+                    init_w as i32,
+                    init_h as i32,
+                    (init_w * 4) as i32,
+                    wl_shm::Format::Argb8888,
+                    qh,
+                    (),
+                );
+                state.buffer = Some(buffer.clone());
+            } else if interface == wl_seat::WlSeat::interface().name {
+                registry.bind::<wl_seat::WlSeat, _, _>(name, version, qh, ());
+            } else if interface == xdg_wm_base::XdgWmBase::interface().name {
+                let wm_base = registry.bind::<xdg_wm_base::XdgWmBase, _, _>(name, 1, qh, ());
+                state.wm_base = Some(wm_base);
+            }
+        }
+    }
+}
+
+impl Dispatch<wl_compositor::WlCompositor, ()> for State {
+    fn event(
+        _: &mut Self,
+        _: &wl_compositor::WlCompositor,
+        _: wl_compositor::Event,
+        _: &(),
+        _: &Connection,
+        _: &QueueHandle<Self>,
+    ) {
+        // wl_compositor has no event
+    }
+}
```

Comment:
Want to use `delegate_noop!` here instead?

```suggestion
delegate_noop!(State: wl_compositor::WlCompositor);
```

Since this is an example, let's use the approach that's simplest for other to follow :)

Reply:
but I want it be the same as xdg-shell, emm, to show they are similar

Reply:
Might be best to update that example to use `delegate_noop!` too.

Reply:
ok, done